### ISSUE TITLE
Calling Deno.exit most probably will prevent coverage collection

### ISSUE THREAD

**Issue by caspervonb** [denoland/deno#8713]
Since Deno.exit calls exit(2) directly, coverage will probably not be collected in that scenario. We'd have to set up an atexit equivalent hook.

**Comment by bartlomieju** [comment]
@caspervonb is this still actionable given we now have `exitSanitizer`?

**Comment by caspervonb** [comment]
> @caspervonb is this still actionable given we now have `exitSanitizer`?

Oh completely forgot about this.
Yeah it still applies to integration tests that are run with `deno run` via `Deno.run`.

**Comment by bartlomieju** [comment]
> Yeah it still applies to integration tests that are run with deno run via Deno.run.

So we'd have to add some kind of call into `op_exit` to prepare a flush of collected coverage?

**Comment by caspervonb** [comment]
> > Yeah it still applies to integration tests that are run with deno run via Deno.run.
> 
> So we'd have to add some kind of call into `op_exit` to prepare a flush of collected coverage?

Yes, altho would be preferable to have something like `atexit(1)` in Rust that can take a closure, that way we can keep it out of the ops.

**Comment by bartlomieju** [comment]
> > > Yeah it still applies to integration tests that are run with deno run via Deno.run.
> > 
> > 
> > So we'd have to add some kind of call into `op_exit` to prepare a flush of collected coverage?
> 
> Yes, altho would be preferable to have something like `atexit(1)` callbacks in Rust that can take a closure, that way we can keep it out of ops.

Not sure if that would work - you still need to access `ProgramState` and `Worker` to finish coverage collection?

I think this might be a non-issue and working as intended unless Node can collect coverage if it process exits.

**Comment by bartlomieju** [comment]
Related to https://github.com/denoland/deno/issues/12888
